### PR TITLE
[pipeline] Tokenizer should not add special tokens for text generation

### PR DESCRIPTION
**Pull request by patrickvonplaten**
This PR fixes generation in pipelines for all models whose tokenizer adds special tokens to the input, *e.g.* XLNet. 

I think this is good for now, but I think the `_parse_and_tokenize()` function needs a larger refactoring to allow more flexibility in the future, also see Issue: #4501

### CODE REVIEW

File: src/transformers/pipelines.py [hunk 1]
```diff
@@ -454,14 +454,17 @@ def ensure_tensor_on_device(self, **inputs):
         """
         return {name: tensor.to(self.device) for name, tensor in inputs.items()}
 
-    def _parse_and_tokenize(self, *args, pad_to_max_length=True, **kwargs):
+    def _parse_and_tokenize(self, *args, pad_to_max_length=True, add_special_tokens=True, **kwargs):
```
Comment (patrickvonplaten) [inline src/transformers/pipelines.py:457]: Currently, all inputs are tokenized with the option `add_special_tokens=True`. This should not be the case for text generation though. `XLNet` for example appends the tokens <seep> and <cls> to the input which leads to bad generation quality. See:
```python
from transformers import XLNetTokenizer
tok = XLNetTokenizer.from_pretrained("xlnet-base-cased") 
tok.decode(tok.encode("Hello"))  # gives "Hello<sep><cls>"
```

File: src/transformers/pipelines.py [hunk 2]
```diff
@@ -617,9 +620,11 @@ def __call__(
             # Manage correct placement of the tensors
             with self.device_placement():
                 if self.model.__class__.__name__ in ["XLNetLMHeadModel", "TransfoXLLMHeadModel"]:
-                    inputs = self._parse_and_tokenize(self.PADDING_TEXT + prompt_text, pad_to_max_length=False)
+                    inputs = self._parse_and_tokenize(
```
Comment (patrickvonplaten) [inline src/transformers/pipelines.py:623]: In generation we should set `add_special_tokens = False`.

**Review by julien-c — APPROVED** [review]
seems very reasonable to me

**Review by LysandreJik — APPROVED** [review]
LGTM

**Pull request merged**